### ISSUE TITLE
how to generates the vocab.txt file from corpus if I want do pre-training from scratch

### ISSUE THREAD

**Issue by loveJasmine** [google-research/bert#533]

**Comment by jeandalf** [comment]
Hello, I believe Google is not sharing the code for this. But the algorithm is described in details in https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/37842.pdf.

Good luck.

**Comment by dsindex** [comment]
@loveJasmine 

https://github.com/google/sentencepiece

i think that you can build a `BPE vocab` via this module.

**Comment by maggieezzat** [comment]
Did you find an answer to that?

**Comment by MaveriQ** [comment]
Is it possible to access/download the standard 30000 token original BERT vocab?

**Comment by raj5287** [comment]
@MaveriQ yes it is just download zip file of any model and you wil find the vocab.txt file there.


> Did you find an answer to that?

Did you?
